framework/source/class/qx/ui/form/VirtualSelectBox.js

```diff
@@ -445,6 +467,150 @@ qx.Class.define("qx.ui.form.VirtualSelectBox",
       var d = event.getData();
       var old = (d.removed.length ? d.removed[0] : null);
       this.fireDataEvent("changeValue", d.added[0], old);
+    },
+
+
+    /*
+    ---------------------------------------------------------------------------
+      INCREMENTAL SEARCH
+    ---------------------------------------------------------------------------
+    */
+
+    __filterValue : null,
+    __lastMatch : '',
+    __filterUpdateRunning : 0,
+    __filterInput : null,
+    __highlightStyle : null,
+    __lastRich : false,
+
+
+     __addFilterInput : function()
+    {
+      var input = this.__filterInput = new qx.ui.form.TextField().set({
+          appearance : 'widget',
+          liveUpdate : true,
+          height     : 0,
+          width      : 1 // must be > 0
+      });
+      this._add(input);
+      input.addListener("focus", function(e) {
+          // reopen after focus loss
+          this.open();
+      }, this);
+      this.addListener("focus", function(e) {
+          // move focus into TextField
+          input.focus();
+      }, this);
+      input.addListener('changeValue', function(e) {
+          if (this.__filterUpdateRunning === 0) this.__updateDelegate();
+      }, this);
+      input.addListener("keypress", function(e) {
+          switch (e.getKeyIdentifier()) {
+          case 'Enter':
+          case 'Escape':
+              this.__updateDelegate();
+              break;
+          // prevent cursor from being moved to beginning/end of input field
+          case 'Down':
+          case 'Up':
+              e.preventDefault();
+              break;
+          }
+      }, this);
+    },
+
+    __applyMarkHighlightStyle : function(value, old) {
+      var styles = [];
+      if (value != null) {
+        var keys = Object.keys(value);
```

Comment:
What do you think about adding the option for highlight style to be a string that maps to an appearance in the theme, since this is a fully themeable widget?

Reply:
That would make sense. Could you point me to an example?

Reply:
since the highlight is created using a span in html with the 'rich' property set on the value ... this does not mesh well with qooxdoo theming

Reply:
Ah. yes, we are only highlighting part of the label string of the list item.

Reply:
The way to pull this from the theme is as so:
```
qx.theme.manager.Appearance.getInstance().styleFrom("list-search-highlight")
```

Reply:
(And then we should add it to the few internal themes we ship with)

Reply:
As stated in the "conversation", you can do it like this:

```
qx.theme.manager.Appearance.getInstance().styleFrom("list-search-highlight")
```
and then add "list-search-highlight" to the few shipped themes.